tests/gauge_alg_test.cpp

```diff
@@ -42,30 +39,31 @@ class GaugeAlgTest : public ::testing::Test {
 
   }
 
-  bool checkDimsPartitioned(){
-    if(comm_dim_partitioned(0) || comm_dim_partitioned(1) || comm_dim_partitioned(2) || comm_dim_partitioned(3)) return true;
+  bool checkDimsPartitioned()
+  {
+    if (comm_dim_partitioned(0) || comm_dim_partitioned(1) || comm_dim_partitioned(2) || comm_dim_partitioned(3))
+      return true;
     return false;
   }
 
   bool comparePlaquette(double3 a, double3 b){
     double a0,a1,a2;
-    a0 = DABS(a.x - b.x);
-    a1=DABS(a.y - b.y);
-    a2=DABS(a.z - b.z);
+    a0 = std::abs(a.x - b.x);
+    a1 = std::abs(a.y - b.y);
+    a2 = std::abs(a.z - b.z);
     double prec_val = 1.0e-5;
-    if(prec == QUDA_DOUBLE_PRECISION) prec_val = 1.0e-15;
-    if( (a0 < prec_val) && (a1  < prec_val)  && (a2  < prec_val) ) return true;
+    if (prec == QUDA_DOUBLE_PRECISION) prec_val = 1.0e-15;
+    if ((a0 < prec_val) && (a1 < prec_val) && (a2 < prec_val)) return true;
     return false;
   }
 
   bool CheckDeterminant(double2 detu){
     double prec_val = 5e-8;
-    if(prec == QUDA_DOUBLE_PRECISION) prec_val = 1.0e-15;
-    if(DABS(1.0 - detu.x) < prec_val && DABS(detu.y) < prec_val) return true;
+    if (prec == QUDA_DOUBLE_PRECISION) prec_val = 1.0e-15;
+    if (std::abs(1.0 - detu.x) < prec_val && std::abs(detu.y) < prec_val) return true;
     return false;
   }
 
-
   void CallUnitarizeLinks(cudaGaugeField *cudaInGauge){
     unitarizeLinks(*cudaInGauge, num_failures_dev);
     cudaMemcpy(&num_failures, num_failures_dev, sizeof(int), cudaMemcpyDeviceToHost);
```

Comment:
Fixed in https://github.com/lattice/quda/pull/1046/commits/74c730d46bc65883bdfa47b5b05e10d65aeb27d5.  I've cleaned up the test code, removing all use of `cudaMemcpy`, `cudaMemset` and `cudaMalloc`.  Moreover, I've also removed a bunch of unnecessary error checking in the test code (which is unneeded since error checking is done internally now) in https://github.com/lattice/quda/pull/1046/commits/3e8642f32fdecd0e3a02522e43b629521bc81c47.